forte/data/readers/openie_reader.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2019 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The reader that reads Open-IE extractions data into data pack.
+Format:
+https://github.com/gabrielStanovsky/oie-benchmark/tree/master/oie_corpus
+"""
+import logging
+import os
+from typing import Iterator, Any
+
+from forte.common.exception import ProcessorConfigError
+from forte.common.configuration import Config
+from forte.common.resources import Resources
+from forte.data.data_utils_io import dataset_path_iterator
+from forte.data.data_pack import DataPack
+from forte.data.readers.base_reader import PackReader
+from ft.onto.base_ontology import Sentence, PredicateMention, \
+    PredicateArgument, PredicateLink, Document
+
+__all__ = [
+    "OpenIEReader"
+]
+
+
+class OpenIEReader(PackReader):
+    r""":class:`OpenIEReader` is designed to read in the Open IE dataset used
+        by Open Information Extraction task. The related paper can be found
+        `here
+        <https://gabrielstanovsky.github.io/assets/papers/emnlp16a/paper.pdf>`__.
+        The related source code for generating this dataset can be found
+        `here
+        <https://github.com/gabrielStanovsky/oie-benchmark>`__.
+        To use this Reader, you must follow the dataset format described
+        `here
+        <https://github.com/gabrielStanovsky/oie-benchmark/tree/master/oie_corpus>`__.
```

Comment:
OK, but it will be good if you explain the data format assumption just a bit. For example, each line contains what. In the perfect scenario, the user may create a dataset himself, without referring to the links you provided.

Reply:
Added. Thanks!